parsing/location.ml

```diff
@@ -463,14 +463,18 @@ let highlight_quote ppf
         (* Single-line error *)
         Format.fprintf ppf "%s | %s@," line_nb line;
         Format.fprintf ppf "%*s   " (String.length line_nb) "";
-        for pos = line_start_cnum to rightmost.pos_cnum - 1 do
+        String.iteri (fun i car ->
+          let pos = line_start_cnum + i in
           if ISet.is_start iset ~pos <> None then
             Format.fprintf ppf "@{<%s>" highlight_tag;
           if ISet.mem iset ~pos then Format.pp_print_char ppf '^'
-          else Format.pp_print_char ppf ' ';
+            (* For alignment purposes, align using a tab for each tab in the
+               source code *)
+          else if car = '\t' then Format.pp_print_char ppf '\t'
+          else if pos < rightmost.pos_cnum then Format.pp_print_char ppf ' ';
```

Comment:
shouldn't both spaces and tabs be placed under the `pos < rightmost.pos_cnum` test?

Reply:
You are right. Fixed.